x-pack/legacy/plugins/ml/server/new_platform/licence_check_pre_routing_factory.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  KibanaRequest,
+  KibanaResponseFactory,
+  RequestHandler,
+  RequestHandlerContext,
+} from 'src/core/server';
+import { PLUGIN_ID } from './plugin';
+
+export const licensePreRoutingFactory = (
+  xpackMainPlugin: any,
+  handler: RequestHandler<any, any, any>
+): RequestHandler<any, any, any> => {
```

Comment:
Should use proper type forwarding here, will be useful when your handlers will be migrated to TS (see `src/core/server/http/router/error_wrapper.ts` for an example)

```typescript
export const licensePreRoutingFactory = <
  P extends ObjectType,
  Q extends ObjectType,
  B extends ObjectType
>(
  xpackMainPlugin: any,
  handler: RequestHandler<P, Q, B>
): RequestHandler<P, Q, B> => {
...
}
```